src/parser/rogue/shared/shadowlands/legendaries/InvigoratingShadowdust.tsx

```diff
@@ -0,0 +1,56 @@
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Abilities from 'parser/core/modules/Abilities';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELLS from 'common/SPELLS';
+import Events, { CastEvent } from 'parser/core/Events';
+
+import { ASSASSINATION_ABILITY_COOLDOWNS, SUBTLETY_ABILITY_COOLDOWNS, OUTLAW_ABILITY_COOLDOWNS, SHARED_ABILITY_COOLDOWNS } from 'parser/rogue/shared/constants';
+import { SpellList } from 'common/SPELLS/Spell';
+
+/**
+ * The Inigorating Shadowdust legendary reduces the cooldown of ALL abilities by 15 seconds after using Vanish. This is usable by all 3 Rogue specs.
+ * 
+ * 10/20/2020 -- Have not been able to find any logs to test this against during the current Beta. Will implement and modify trackable statistic(s), if any, when there are logs available.
+ */
+
+class InvigoratingShadowdust extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+
+  protected spellUsable!: SpellUsable;
+  spec: string = '';
+  cooldownReduction: number = 15000; // 15 seconds
+
+  constructor(options: Options) {
+    super(options);
+    this.spec = this.selectedCombatant.spec().specName;
+    this.active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.INVIGORATING_SHADOWDUST.bonusID);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VANISH), this.onCast);
+  }
+
+  onCast(event: CastEvent) {
+    let cooldowns: SpellList[];
+    switch (this.spec) {
+      case 'Assassination':
```

Comment:
I would populate the cooldowns list in the constructor so it doesn't run every single onCast event - there's no reason for the switch to run every single time either. 
also you could do: 
```js 
let cooldowns: SpellList[] = [SHARED_ABILITY_COOLDOWNS]
```

Followed by a 
```js
cooldowns.push(SPEC_ABILITY_COOLDOWNS)
```